src/parser/type_checking.c

```diff
@@ -20363,6 +20401,19 @@ pt_semantic_type (PARSER_CONTEXT * parser, PT_NODE * tree, SEMANTIC_CHK_INFO * s
       tree = NULL;
     }
 
+  /* When pt_lambda_node is executed in mq_optimize, a removable predicate like '1=1' is generated.
```

Comment:
pt_lambda_node() is a general function to exchange pt_node. qo_reduce_equality_terms() is a more specific function.

Reply:
I will change the function name mentioned in the comments to qo_reduce_equality_terms.